chcp_beforeInstall event

toostn opened the pull request:
I've implemented and tested this for iOS. Event is triggered correctly, and I can use it for showing the splash screen as desired in #100.

For Android, maybe the easiest way to fix this is to trigger the event before the InstallationWorker runs?

nikDemyankov commented:
Thanks!
As I said before - Android is implemented in a very similar way, as iOS. So yes, probably just need to dispatch the event either in [UpdatesInstaller](https://github.com/nordnet/cordova-hot-code-push/blob/master/src/android/src/com/nordnetab/chcp/main/updater/UpdatesInstaller.java#L71) or [InstallationWorker](https://github.com/nordnet/cordova-hot-code-push/blob/master/src/android/src/com/nordnetab/chcp/main/updater/InstallationWorker.java#L56).

toostn commented:
I've added my attempt of an Android implementation. This works well for my use case and doesn't seem to interfere with anything else, but please check it before merging :)

Since I don't know if the EventBus is thread safe, I added the event trigger outside of the InstallationWorker thread, right before the worker is executed. Feel free to move it wherever you see it fits.

nikDemyankov commented:
Awesome, thanks! Seems to be fine. I will merge it in branch dev/v1.3.0, since it's a new feature.

nikDemyankov commented:
Merged in branch dev/v1.3.0 with minor bugfix.
